### PR TITLE
add admin E2E tests

### PR DESCRIPTION
**Pull request by rrennick**
### Changes proposed in this Pull Request:

This PR updates the core E2E test run to include the tests in `@woocommerce/admin-e2e-tests`. It removes the duplicate tests that exist in `core-tests` from the test run.

### How to test the changes in this Pull Request:

1. Run `composer install --no-dev`
2. Run `npm i`
3. Run E2E tests locally
4. Re-run tests with `E2E_RETEST=1`

### Changelog entry

> N/A

### CODE REVIEW

**Comment by rrennick** [comment]
@zhongruige Locally I see an `Node is either not visible or not an HTMLElement` error on the `Analytics pages › should allow a user to add a section back in` every time on this branch. The screenshot shows that the element clicked in the test is visible.

I believe that is due to differences between the Chromium version for Mac and Linux. The WC Admin repo uses a newer version of Puppeteer which seems to addressed the issue. To update Puppeteer and dependencies in this repo we need to update `@automattic/puppeteer-utils` as well. I'd like to address that in separate PRs. If you encounter the same error locally, we can prioritize the follow up PRs.

**Comment by rrennick** [comment]
> The WC Admin repo uses a newer version of Puppeteer which seems to address the issue. 

I just tested in the WC Admin repo and see the same error reported there when running locally.

**Comment by zhongruige** [comment]
> @zhongruige Locally I see an `Node is either not visible or not an HTMLElement` error on the `Analytics pages › should allow a user to add a section back in` every time on this branch. The screenshot shows that the element clicked in the test is visible.
> 
> I believe that is due to differences between the Chromium version for Mac and Linux. The WC Admin repo uses a newer version of Puppeteer which seems to addressed the issue. To update Puppeteer and dependencies in this repo we need to update `@automattic/puppeteer-utils` as well. I'd like to address that in separate PRs. If you encounter the same error locally, we can prioritize the follow up PRs.

This seemed to work for me, so we may not need to prioritize this now, though we may need dig into this a bit more if other folks start seeing it:

Screen Shot 2021-08-25 at 14 39 52

File: tests/e2e/env/utils/update-ready-page.js [hunk 1]
```diff
@@ -25,11 +25,11 @@ const updateReadyPageStatus = async ( status ) => {
 		statusFilter = 'draft';
 	}
 
-	const getPostsResponse = await client.get(`${wpPagesEndpoint}?search=ready&status=${statusFilter}`);
-	if ( getPostsResponse.data && getPostsResponse.data.length > 0) {
+	const getPostsResponse = await client.get(`${wpPagesEndpoint}?search=ready&status=${statusFilter}` );
```
Comment (zhongruige) [inline tests/e2e/env/utils/update-ready-page.js]: Minor changes here to add spacing to the front:

```suggestion
	const getPostsResponse = await client.get( `${wpPagesEndpoint}?search=ready&status=${statusFilter}` );
```

File: tests/e2e/env/utils/update-ready-page.js [hunk 2]
```diff
@@ -25,11 +25,11 @@ const updateReadyPageStatus = async ( status ) => {
...
+	if ( getPostsResponse.data && getPostsResponse.data.length > 0 ) {
 		const pageId = getPostsResponse.data[0].id;
 		// Update the page to the new status
-		await client.post(`${wpPagesEndpoint}/${pageId}`, { 'status': status });
+		await client.put(`${wpPagesEndpoint}/${pageId}`, { 'status': status } );
```
Comment (zhongruige) [inline tests/e2e/env/utils/update-ready-page.js]: Curious on the change from the `post` to `put` here. The [WordPress API specifies using a `post` for the update](https://developer.wordpress.org/rest-api/reference/pages/#update-a-page), but it looks like both are supported (`put` definitely makes more sense for an update action in an API, so just trying to understand a bit more around the change, or if it's just for better REST API semantics).

Add spacing:

```suggestion
		await client.put( `${wpPagesEndpoint}/${pageId}`, { 'status': status } );
```

File: tests/e2e/core-tests/specs/merchant/wp-admin-product-new.test.js [hunk 3]
```diff
@@ -242,15 +221,7 @@ const runAddVariableProductTest = () => {
...
-			/**/
-
-			// Publish product, verify that it was published. Trash product, verify that it was trashed.
-			await verifyPublishAndTrash(
-				'#publish',
-				'.notice',
-				'Product published.',
-				'1 product moved to the Trash.'
-			);
+			// @todo: add one or more tests to verify changes were saved
```
Comment (zhongruige) [inline tests/e2e/core-tests/specs/merchant/wp-admin-product-new.test.js]: Added a follow up issue for this: https://github.com/woocommerce/woocommerce/issues/30580

**Review by zhongruige — DISMISSED** [review]
Looks great @rrennick! These ran great for me locally, and I didn't see the test failure mentioned in the comments.

Approved for now, just a few minor comments on spacing. I did want to double check since it isn't added here, is this something we'll should add a changelog and README entry for? If not, this is good to merge once the comments are addressed.

For the `Analytics pages › should allow a user to add a section back in` test, I added in this follow-up task so we can further investigate: https://github.com/woocommerce/woocommerce/issues/30581

File: tests/e2e/env/utils/update-ready-page.js (hunk 2 above)
Comment (rrennick) [reply tests/e2e/env/utils/update-ready-page.js]: > just trying to understand a bit more around the change, or if it's just for better REST API semantics

Mainly semantics. With the Woo REST API it has to be `PUT`. I think WP may have included support for `POST` due to some hosting providers not supporting `PUT` by default.

**Comment by rrennick** [comment]
> I did want to double check since it isn't added here, is this something we'll should add a changelog and README entry for?

This PR doesn't make any changes to any of the packages other than updating `POST` to `PUT`. I suppose we could add an entry for that though.

> This seemed to work for me, so we may not need to prioritize this now

This could be something local to my machine. We can compare notes on the follow up issue.

File: tests/e2e/env/utils/update-ready-page.js (hunk 2 above)
Comment (zhongruige) [reply tests/e2e/env/utils/update-ready-page.js]: Oh gotcha, thanks for the clarification!

**Comment by zhongruige** [comment]
> This PR doesn't make any changes to any of the packages other than updating POST to PUT. I suppose we could add an entry for that though.

Oh totally makes sense, I think that is okay to leave out.

> This could be something local to my machine. We can compare notes on the follow up issue.

Sounds good!

**Comment by rrennick** [comment]
> This PR doesn't make any changes to any of the packages other than updating POST to PUT.

@zhongruige Thanks for the feedback. I changed this back to a `POST` since that's the method used in the documentation. I updated for the feedback and removed a function that was a duplicate and not being used. If you don't mind, look it over for me again.

**Review by zhongruige — APPROVED** [review]
Changes look great! Thanks for getting this in @rrennick and cleaning up the unused function as well. Let's get this merged in :shipit:

**Pull request merged**